pymc3/step_methods/metropolis.py

```diff
@@ -120,9 +120,10 @@ def astep(self, q0):
                 q = (q0 + delta).astype(int)
             else:
                 delta[self.discrete] = round(delta[self.discrete], 0).astype(int)
-                q = q0 + delta
+                q = (q0 + delta)
         else:
-            q = q0 + delta
+            q0 = q0.astype(theano.config.floatX)
+            q = (q0 + delta).astype(theano.config.floatX)
```

Comment:
Does this effect numerical precision when theano is not actually needed? Can we use `model.dtype`instead of `theano.config.floatX`? This way if the model does not actually need theano or to run on GPU, the sampling can still run in `float64` even if `theanorc` specifies `float32`.